src/web_service/cookie.cpp

```diff
@@ -0,0 +1,1051 @@
+// SPDX-FileCopyrightText: 2023 Yuzu Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <sstream>
+
+#include "common/string_util.h"
+#include "cookie.h"
+
+namespace WebService {
+
+#define TM_YEAR_BASE 1900
+
+Cookie::Cookie(std::string& name, std::string& value) : m_name{name}, m_value{value} {}
```

Comment:
Pass by value and move

```suggestion
Cookie::Cookie(std::string name, std::string value) : m_name{std::move(name)}, m_value{std::move(value)} {}
```

this allows the caller to either copy if it needs to, or move into the cookie, completely eliminating copies altogether, making the interface way more flexible